### ISSUE TITLE
ESP8266 using Sleep as Android

### ISSUE THREAD

**Issue by MaxiBerdy** [probonopd/ESP8266HueEmulator#84]
Hello there,
I'm trying to know if there is a way the app Sleep as Android could recognize the hub created with the ESP8266 as an actual Philips Hue Hub, since there's a functionality to use philips smartbulbs in this app. This way, I could plug a ws2812b led strip on this "false" hub and then make it recognize on this app, and wakeup with controlled ambient lightning with it...
Do you know if this could work ?

Thanks for any answers or advices !
(Sorry for my English !)

**Comment by Carstijn** [comment]
Hey MaxiBerdy,

I use the hueemulator for some time in combination with sleep as android, and it preforms very well. I trust is so much I don't even have a alarm with sound setup. (I use a hacked up H801 led controller with oldschool RGB pwm led strip)

So I hope it works out for you!

**Comment by MaxiBerdy** [comment]
Oh thanks for the answer ! So do you mean the HueEmulator is easily recognized by the Sleep for Android App ? Then I might consider doing so !
Was it easy to make ?

**Comment by Carstijn** [comment]
I guess it is detect as a normal hue bridge (I do not own an original). The smartlight features seem to work including the test function/sun rise wake up/ and nightlight function within sleep as android. Was is hard? I guess at this point it is just uploading the sketch and connect some led's ;-)

**Comment by MaxiBerdy** [comment]
Oh well perfect, you convinced me then I guess ! Only problem is that you have RGBW led and I got ws2812b leds which are "+, Data, Ground" ones.. Not sure if it will control those leds, but i'll see and search a bit more ! Thanks dude

**Comment by probonopd** [comment]
@Carstijn I guess we would have to implement this functionality in ESP8266HueEmulator but since hueemulator already does it it should be doable, correct?

**Comment by MaxiBerdy** [comment]
hueemulator is an other thing ? I thought it was the same as ESP8266HueEmulator.. I'm lost now !

**Comment by probonopd** [comment]
@Carstijn which "hueemulator" were you talking about? (I am not sure where the timers are implemented but I think they need to be implemented in the emulated bridge?)

**Comment by Carstijn** [comment]
I was talking about the ESP8266HueEmulator, which works just fine with sleep as android. (at least for me I does)

**Comment by probonopd** [comment]
Thanks for the clarification, good to know. Does that mean the Android app locally manages the timer, and sends a command to the bridge when the time has come to wake you up? (What if the WLAN is down at that time?)

**Comment by Carstijn** [comment]
Yes, sleep as android does turn the lights on when its time to wake. I guess the scene support is not in place at the moment.

For the wlan being offline, I'm more scared of not pulling in the charger and waking up to a dead battery ;-)

**Comment by udavst** [comment]
Sleep as Android does not work very well, it regulates the light irregularly, but this is the problem of Sleep As Android.
I do not understand why, after turning off the alarm, the light remains on.
